librz/core/project.c

```diff
@@ -47,13 +62,32 @@ RZ_API RzProjectErr rz_project_save_file(RzCore *core, const char *file) {
 		sdb_free(prj);
 		return err;
 	}
-	if (!sdb_text_save(prj, file, true)) {
+	if (!sdb_text_save(prj, save_file, true)) {
 		err = RZ_PROJECT_ERR_FILE;
 	}
 	sdb_free(prj);
-	if (err == RZ_PROJECT_ERR_SUCCESS) {
-		rz_config_set(core->config, "prj.file", file);
+
+	if (!compress) {
+		if (err == RZ_PROJECT_ERR_SUCCESS) {
+			rz_config_set(core->config, "prj.file", file);
+		}
+		return err;
 	}
+
+	if (err != RZ_PROJECT_ERR_SUCCESS) {
+		goto tmp_file_err;
+	}
+
+	if (!rz_file_deflate(tmp_file, file)) {
+		err = RZ_PROJECT_ERR_COMPRESSION_FAILED;
+		goto tmp_file_err;
+	}
+
+	rz_config_set(core->config, "prj.file", file);
```

Comment:
```suggestion
	if (err != RZ_PROJECT_ERR_SUCCESS) {
		goto tmp_file_err;
	}
	
	if (compress && !rz_file_deflate(tmp_file, file)) {
		err = RZ_PROJECT_ERR_COMPRESSION_FAILED;
		goto tmp_file_err;
	}

	rz_config_set(core->config, "prj.file", file);
```
something like this.

Reply:
Thanks. This makes it much simpler.